src/BloomExe/Book/Book.cs

```diff
@@ -1104,6 +1122,24 @@ private void UpdatePageFromFactoryTemplates(HtmlDom bookDom, IProgress progress)
 			UpdateMultilingualSettings(bookDom);
 		}
 
+		public void UpdatePageToTemplate(XmlDocument bookDom, XmlElement templatePageDiv, string pageId)
```

Comment:
I've been trying to nudge this kind of detailed xml manipulation out of the Book class where possible. In case you didn't consider it, please consider whether it would make sense to move this to HTMLDom class ("HtmlDom manages the lower-level operations on a Bloom XHTML DOM.")?